src/main/scala/Preparator.scala

```diff
@@ -55,12 +55,12 @@ class Preparator
       (eventName, eventIDS.create(eventIDS.matrix, userDictionary.get, eventIDS.columnIDs).newRowCardinality(numUsers))
     }
 
-    new PreparedData(rowAdjustedIds, trainingData.fieldsRDD)
+    PreparedData(rowAdjustedIds, trainingData.fieldsRDD)
   }
 
 }
 
-class PreparedData(
-    val actions: List[(String, IndexedDataset)],
-    val fieldsRDD: RDD[(String, PropertyMap)])
-  extends Serializable
\ No newline at end of file
+case class PreparedData(
+    actions: List[(String, IndexedDataset)],
+    fieldsRDD: RDD[(String, PropertyMap)]
+) extends Serializable
```

Comment:
1) end-parenthesis does not match Scala-style docs. http://docs.scala-lang.org/style/declarations.html#classes
2) are you sure a case class works, this is not the way templates are usually defined. They use classes.
3) thanks for removing `val`